server/src/main/java/org/opensearch/index/store/remote/utils/TransferManager.java

```diff
@@ -94,27 +96,17 @@ public IndexInput fetchBlob(BlobFetchRequest blobFetchRequest) throws IOExceptio
     }
 
     private IndexInput downloadBlockLocally(BlobFetchRequest blobFetchRequest) throws IOException {
-        // We need to do a privileged action here in order to fetch from remote
-        // and write to the local file cache in case this is invoked as a side
-        // effect of a plugin (such as a scripted search) that doesn't have the
-        // necessary permissions.
-        return AccessController.doPrivileged((PrivilegedAction<? extends IndexInput>) () -> {
-            try {
-                try (
-                    InputStream snapshotFileInputStream = blobContainer.readBlob(
-                        blobFetchRequest.getBlobName(),
-                        blobFetchRequest.getPosition(),
-                        blobFetchRequest.getLength()
-                    );
-                    OutputStream fileOutputStream = Files.newOutputStream(blobFetchRequest.getFilePath());
-                    OutputStream localFileOutputStream = new BufferedOutputStream(fileOutputStream);
-                ) {
-                    snapshotFileInputStream.transferTo(localFileOutputStream);
-                }
-                return blobFetchRequest.getDirectory().openInput(blobFetchRequest.getFileName(), IOContext.READ);
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
-            }
-        });
+        try (
+            InputStream snapshotFileInputStream = blobContainer.readBlob(
```

Comment:
@andrross AFAIK blobContainer.readBlob is always executed under `doPrivileged` blocks ...

Reply:
(each repository-xxx has tests for that)

Reply:
@reta Yes that is correct, but this was failing with the `SpecialPermissions.check()` that happens right before the doPrivileged block within the repository. Then it also fails when the file cache interacts with the file system to write a new block or delete one that is evicted.

Reply:
Oh I see, thanks!